selfdrive/ui/qt/offroad/wifiManager.cc

```diff
@@ -387,10 +380,28 @@ void WifiManager::propertyChange(const QString &interface, const QVariantMap &pr
       knownConnections = listConnections();
     }
     refreshNetworks();  // TODO: only refresh on first scan, then use AccessPointAdded and Removed signals
+    updateUI();
+
+  } else if (interface == wireless_device_iface && props.contains("ActiveAccessPoint")) {
+    const QDBusObjectPath &path = props.value("ActiveAccessPoint").value<QDBusObjectPath>();
+    activeAp = path.path();  // can be "/" or empty if no active AP
+    tetheringEnabled = getApProperty(activeAp, "Ssid") == tethering_ssid;
+  }
+}
+
+void WifiManager::updateUI() {
+  if (this->isVisible()) {
     emit refreshSignal();
   }
 }
 
+void WifiManager::deviceAdded(const QDBusObjectPath &path) {
+  if (isWirelessAdapter(path)) {
```

Comment:
can this be run multiple times?

Reply:
Yes, there's another adapter on PC that's called `NM_DEVICE_TYPE_WIFI_P2P` that's added at the same time

Reply:
shouldn't this only be called once?

Reply:
`deviceAdded`? It's called twice when the WiFi adapter is plugged in, I'll check the device

Reply:
Made it so that it will only get the adapter once, which if we're certain it never is removed is okay (it's a different path each time it's added)

Reply:
I just added a commit to make it resilient to losing the adapter (equivalent to unplugging and replugging in the USB on PC) by reconnecting the signals using the old adapter path. If we don't care about this, I can revert the commit.